starknet_devnet/devnet_config.py

```diff
@@ -276,6 +276,10 @@ def parse_args(raw_args: List[str]):
         type=_fork_block,
         help="Specify the block number where the --fork-network is forked; defaults to latest",
     )
+    parser.add_argument(
+        "--chain-id",
+        help="Specify the chain id as string, MAINNET or TESTNET",
```

Comment:
1. Some validation should be done here, similar to e.g. `_fork_block` or `_fork_network` (there might be a small bug in the latter).
2. The help string doesn't have to hardcode legal values, it can use f-strings like in `--fork-network`.
3. You can set the default value here, and not in the `build_config` function (see e.g. `--timeout` for reference). The `build_config` function can then work without an optional argument.

Reply:
validation https://github.com/Shard-Labs/starknet-devnet/pull/358/commits/406847467663b09777a15a7746aa9e1a7f44f5f4

tests added in https://github.com/Shard-Labs/starknet-devnet/pull/358/commits/dc381ff89ba81ff1650365d5de285e4d71aafcad